src/west/commands/help.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''West common commands'''
+
+from west.commands import WestCommand
+from west.util import WestReload
+
+class Help(WestCommand):
+    def __init__(self):
+        super().__init__(
+            'help',
+            'print west command help text',
+            'Print help text on any command.'
+        )
+
+    def do_add_parser(self, parser_adder):
+        return self.add_parser_helper(
+            parser_adder,
+            self.add_arg('command', metavar='cmd', nargs='?',
+                         help='Command to get help on')
+        )
+
+    def do_run(self, args, user_args):
+        a = [args.command, '-h']
+        wr = WestReload(argv=[x for x in a if x is not None])
```

Comment:
Testing this out, I noticed that if an invalid command name is given, then user gets the default `west` usage.
```
$ west help unknown_cmd
usage: west [-h] [-z ZEPHYR_BASE] [-v] [-V] <command> ...
west: error: argument <command>: invalid choice: 'unknown_cmd' (choose from 'list', 'manifest', 'diff', 'status', 'update', 'selfupdate', 'forall', 'post-init', 'help', 'build', 'flash', 'd
```
But in this case, `help` which is a valid command was given.

In contrast , if too many args a given to `west help` you'll get a `west help` error message.
```
$ west help list list
usage: west help [-h] [cmd]
west help: error: unexpected arguments: ['list']
```

Unfortunately, i'm not sure this is so easy to fix.

Reply:
Yeah, I noticed that too. We can fix this later but it's not going to be easy because of the way we use ArgParse. It's `parse_known_ags` that deals with that automagically for us and displays the help text.